### ISSUE TITLE
`--no-deps` flag for `poetry add`

### ISSUE THREAD

**Issue by caniko** [python-poetry/poetry#5900]
- [x] I have searched the [issues](https://github.com/python-poetry/poetry/issues) of this repo and believe that this is not a duplicate.
- [x] I have searched the [documentation](https://python-poetry.org/docs/) and believe that my question is not covered.

## Feature Request

I am in a situation where my suite consists of two packages (X and Y), where `package A` is required in both. The twist is that `package X`  only requires `package A`, and doesn't need its dependencies.

I would like to `poetry add <package>` without dependencies in my poetry package, this is currently not possible.

In other words, I am looking for an equivalent flag `--no-deps` from `pip install` in `poetry add`.

**Comment by finswimmer** [comment]
Hello @caniko,

thank you for your suggestion. I don't think this is something we will implement in Poetry. Unlike `pip`, Poetry isn't a general purpose installation tool. It's a tool that handle your dependencies and build proper python packages. In the Python ecosystem there is no way to say: "My package depends on package A, but not on its dependencies".

If your really think, that in your case `package A` runs without any of its dependencies, you could contact the maintainer and ask if it would make sense to define its dependencies within an `extras` group. Those are optional and must be selected explicit on installation.

fin swimmer

**Comment by caniko** [comment]
Sounds good. It is good to document the maintainers' opinion. Cheers.
